src/fw_update.c

```diff
@@ -182,16 +182,22 @@ static bool received_new_target_component(const struct golioth_ota_manifest *man
             return true;
         }
     }
+    else
+    {
+        GLTH_LOGI(TAG,
+                  "Manifest does not contain target component: %s",
+                  ctx->config.fw_package_name);
+        /* TODO: Report state/reason here.
+         *  This can't be done directly because it would call a sync func from a callback
+         *  Consider adding a new reason code: GOLIOTH_OTA_REASON_COMPONENT_NOT_FOUND
+         */
+    }
+
     return false;
 }
 
-static void fw_report_and_observe(void)
+static void fw_register_observe(void)
```

Comment:
This function name feels a little awkward. Maybe `fw_register_manifest_observation()` or `fw_observe_manifest()`?